### PR TITLE
The Crossbow

### PR DESCRIPTION
**Pull request by Sibg**
## About The Pull Request

This adds the X-8 Crossbow, what hopefully will be in the future a orderable crossbow, for now it justs *exists*

## Why It's Good For The Game

Adds a quite different weapon, capable of firing fire bolts, explosive bolts and its standard amour piercing bolts, did I mention its natively silent?

## Changelog
:cl:
add: Crossbow, It's ammo types.
imageadd: Sprites for the crossbow.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/magazines/shotguns.dm [hunk 1]
```diff
@@ -80,6 +80,7 @@ Generic internal magazine. All shotguns will use this or a variation with differ
 Since all shotguns share ammo types, the gun path is going to be the same for all of them. And it
 also doesn't really matter. You can only reload them with handfuls.
 */
+
 /obj/item/ammo_magazine/internal/shotgun
 	name = "shotgun tube"
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/shotguns.dm:85]:
```suggestion
*/
/obj/item/ammo_magazine/internal/shotgun
	name = "shotgun tube"
```

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -1097,6 +1097,68 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	hud_state_empty = "bolt_empty"
+	damage_falloff = 0
+	shell_speed = 2
+	accuracy = 40
+	damage = 70
+	penetration = 40
+	sundering = 10
+
+/datum/ammo/bullet/shotgun/bolt/ib
+	name = "napalm bolt"
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1120]: Better name than IB please

File: code/modules/projectiles/magazines/specialist.dm [hunk 3]
```diff
@@ -99,6 +99,29 @@ obj/item/ammo_magazine/rifle/m4ra/smart
...
+	caliber = "12g"
+	icon_state = "crossbow piercing bolt"
+	current_rounds = 5
+	default_ammo = /datum/ammo/bullet/shotgun/bolt
+
+/obj/item/ammo_magazine/handful/bolt/ib
+	caliber = "12g"
+	name = "handful of incendiary crossbow bolts"
+	icon_state = "  "
+	current_rounds = 5
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/specialist.dm:116]: ?

File: code/modules/projectiles/magazines/specialist.dm [hunk 4]
```diff
@@ -99,6 +99,29 @@ obj/item/ammo_magazine/rifle/m4ra/smart
...
+	default_ammo = /datum/ammo/bullet/shotgun/bolt/ib
+
+/obj/item/ammo_magazine/handful/bolt/ex
+	caliber = "12g"
+	name = "handful of explosive crossbow bolts"
+	icon_state = "  "
+	current_rounds = 5
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/specialist.dm:123]: ?

**Review by TiviPlus — COMMENTED** [review]
image
Dunno about it being a shotgun but do test that it gives the correct messages and so please

**Comment by Marmio64** [comment]
if this was a bow with explosive arrows, we could do real rambo time

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -1097,6 +1097,68 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	accuracy = 40
+	damage = 70
+	penetration = 40
+	sundering = 10
+
+/datum/ammo/bullet/shotgun/bolt/incendiary_bolts
+	name = "napalm bolt"
+	hud_state = "bolt_fire"
+	flags_ammo_behavior = AMMO_INCENDIARY|AMMO_EXPLOSIVE|AMMO_SUNDERING
+	armor_type = "fire"
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1123]: is it explosive though?

File: code/modules/projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -762,6 +762,39 @@
 	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_POLICE|GUN_WIELDED_FIRING_ONLY|GUN_AMMO_COUNTER
 	req_access = list(ACCESS_MARINE_BRIG)
 
+/*
+* Crossbow
+*/
+
+/obj/item/weapon/gun/shotgun/crossbow
+	name = "\improper X8 Crossbow"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/specialist.dm:770]:
```suggestion
	name = "\improper X8 crossbow"
```

File: code/modules/projectiles/guns/specialist.dm [hunk 7]
```diff
@@ -762,6 +762,39 @@
...
+	desc = "A lightweight X8 Crossbow developed by Heron Hunting, it smells like fresh metal. "
```
Comment (Fikou) [inline code/modules/projectiles/guns/specialist.dm:771]:
```suggestion
	desc = "A lightweight X8 Crossbow developed by Heron Hunting, it smells like fresh metal."
```

File: code/modules/projectiles/magazines/specialist.dm [hunk 8]
```diff
@@ -99,6 +99,29 @@ obj/item/ammo_magazine/rifle/m4ra/smart
 	default_ammo = /datum/ammo/bullet/smartgun/dirty
 	gun_type = /obj/item/weapon/gun/smartgun/dirty
 
+//-------------------------------------------------------
+//CROSSBOW
+
+/obj/item/ammo_magazine/handful/bolt
+	name = "handful of ap crossbow bolts"
+	caliber = "12g"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:107]: 20 inch carbon tipped arrows

File: code/modules/projectiles/magazines/specialist.dm [hunk 9]
```diff
@@ -99,6 +99,29 @@ obj/item/ammo_magazine/rifle/m4ra/smart
...
+	icon_state = "crossbow piercing bolt"
+	current_rounds = 5
+	default_ammo = /datum/ammo/bullet/shotgun/bolt
+
+/obj/item/ammo_magazine/handful/bolt/incendiary_bolts
+	caliber = "12g"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:113]: 20 inch carbon tipped arrows, incendiary-tipped

**Comment by Fikou** [comment]
why add a weapon if youre not gonna put it anywhere seems kinda dumb

File: code/modules/projectiles/magazines/specialist.dm [hunk 10]
```diff
@@ -99,6 +99,29 @@ obj/item/ammo_magazine/rifle/m4ra/smart
...
+	name = "handful of incendiary crossbow bolts"
+	icon_state = "crossbow piercing bolt"
+	current_rounds = 5
+	default_ammo = /datum/ammo/bullet/shotgun/bolt/incendiary_bolts
+
+/obj/item/ammo_magazine/handful/bolt/explosive_bolts
+	caliber = "12g"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/specialist.dm:120]:
```suggestion
	caliber = "carbontippedarrow"
```

File: code/modules/projectiles/magazines/specialist.dm (hunk 8 above)
Comment (Sibg) [reply code/modules/projectiles/magazines/specialist.dm:107]: Its a bolt not a arrow.

**Comment by Sibg** [comment]
Dead forever.

**Pull request closed without merging**